website/common/script/ops/scoreTask.js

```diff
@@ -277,22 +278,45 @@ module.exports = function scoreTask (options = {}, req = {}) {
         task.completed = true;
 
         // Save history entry for daily
+        let dateScored = moment();
+        if (task.yesterDailyScored) {
+          // If this is a yesterDaily being scored, change the date to the last minute of the user's "yesterday"
+          dateScored.endOf('day').subtract({
+            days: 1,
+            minutes: user.preferences.tz - user.preferences.dayStart * 60 + 1,
+          });
+        }
         task.history = task.history || [];
         let historyEntry = {
-          date: Number(new Date()),
+          date: Number(new Date(dateScored)),
           value: task.value,
         };
         task.history.push(historyEntry);
       } else if (direction === 'down') {
+        // Delete history entry when daily unchecked
+        if (task.history || task.history.length > 0) {
+          // @REVIEW @TODO Don't remove entries not from user's "today", in case this is a (single) completion shared daily completed by someone else
+          // task.history.splice(-1, 1);
+          let historyEntry = task.history.pop();
+          if (task.group && task.group.sharedCompletion && task.group.sharedCompletion === 'singleCompletion') {
+            // This Daily could have been completed "today" by someone else.
+            // Since this function runs on the client, as well as the server, let's make assumptions and keep things simple
+            // Otherwise we have to load the Master Task and check who did what when
+            let o = sanitizeOptions(user.preferences);
+            let startOfDayWithCDSTime = startOfDay(o);
+            if (moment(historyEntry.date).isBefore(startOfDayWithCDSTime)) {
+              // This Daily was completed by us before "today"
+              // It was probably completed "today" by someone else
+              task.history.push(historyEntry);
+              throw new NotAuthorized(i18n.t('messageOtherAssigneeCompleted', req.language));
```

Comment:
This catch for "uncompleting" a daily another assigned user completed makes a significant assumption. Please review.

Reply:
Just to make sure it's clear, given the `throw` here the function will stop running and nothing will be saved